### ISSUE TITLE
Missing all loss functions

### ISSUE THREAD

**Issue by agibsonccc** [deeplearning4j/libnd4j#575]
This was only the first one I found:
name: "softmax_cross_entropy_loss/xentropy"
op: "SoftmaxCrossEntropyWithLogits"
input: "softmax_cross_entropy_loss/Reshape"
input: "softmax_cross_entropy_loss/Reshape_1"
attr {
  key: "T"
  value {
    type: DT_FLOAT
  }
}

**Comment by shyrma** [comment]
Hi Adam!
We have implemented following loss functions:
1) absolute difference
2) cosine distance
3) hinge_loss
4) huber loss
5) log loss
6) mean pairwise squared_error
7) mean squared error
8) sigmoid cross entropy
9) softmax cross entropy
During development we were guided by appropriate tf documentation.
